src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/IParameterSymbolExtensions.cs

```diff
@@ -52,5 +52,25 @@ public static bool IsRefOrOut(this IParameterSymbol symbol)
 
             return null;
         }
+
+        public static bool IsPrimaryConstructor(this IParameterSymbol parameter, CancellationToken cancellationToken)
+        {
+            if (parameter is
+                {
+                    DeclaringSyntaxReferences.Length: > 0,
+                    ContainingSymbol: IMethodSymbol
+                    {
+                        MethodKind: MethodKind.Constructor,
+                        DeclaringSyntaxReferences.Length: > 0,
```

Comment:
```suggestion
                        DeclaringSyntaxReferences.Length: [var constructorReference, ..],
```

can then use that below.